util/reflect.go

```diff
@@ -318,19 +319,23 @@ func InsertIntoMapStructField(parentStruct interface{}, fieldName string, key, f
 	return nil
 }
 
+// isFieldTypeCompatible reports whether f.Set(v) can be called successfully on
+// a struct field f corresponding to ft.
```

Comment:
There is `func (reflect.Value) CanSet() bool` -- which checks `CanAddr` and whether the struct field was unexported. I think in our case, there is no such case whereby we can get an unexported field, but perhaps worth checking this too?

Reply:
In this case the field itself is not currently available (only the StructField and value to set the field to) so I'd have to extend the API to pass in the field. It doesn't seem worth it to me given our top level assumption of everything being exported. However, I updated the comments to make these assumptions clear.